src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/InstallPackageCommand.cs

```diff
@@ -46,6 +52,31 @@ protected override void Preprocess()
             ActionType = NuGetActionType.Install;
         }
 
+        // Checks if the sourse is valid http, local or known source. Else throws an exception.
+        private void CheckSourceValidity()
+        {            
+            if (!string.IsNullOrEmpty(Source))
+            {
+                // Convert source into a PackageSource
+                PackageSource packageSource = new PackageSource(Source);
+
+                // Look through all available sources (including those disabled) by matching source name and url
+                var matchingSource = GetMatchingSource(Source);
```

Comment:
GetMatchingSource seems to be getting called twice in this codepath. Try avoiding duplicate calls.